Updated release notes and image version reference docs for 0.23.14 (#332)
astronomer/docs: enterprise/v0.23/09_resources/01_release-notes.md

## v0.23.14

Release Date: April 27, 2021

### Support for Airflow 2.0.2

Astronomer Enterprise v0.23 now offers full support for [Airflow 2.0.2](hhttps://github.com/apache/airflow/releases/tag/2.0.2). Airflow 2.0.2 builds upon the success of Airflow 2.0.0 and 2.0.1 with more bug fixes and performance improvements, including:

- Gracefully handling missing `start_date` and `end_date` for DagRuns ([Source](https://github.com/apache/airflow/pull/14452))
- Faster default role syncing during Webserver start ([Source](https://github.com/apache/airflow/pull/15017))
- Increased Webserver start-up speed when there are many DAGs ([Source](https://github.com/apache/airflow/pull/14993))
- Authenticated plugins endpoints ([Source](https://github.com/apache/airflow/pull/14570))

> For a full list of all changes, read the [Apache Airflow changelog](https://github.com/astronomer/airflow/releases/tag/v2.0.2%2Bastro.1).

Reviewer: ```suggestion
For a list of Astronomer-only changes, read the [Astronomer Certified changelog](https://github.com/astronomer/ap-airflow/blob/master/2.0.2/CHANGELOG.md).
```

You link to OS changelog above, so I think here link out to AC-specific changelog?

> ### Support for Latest Builds of Astronomer Certified

Author: I wasn't sure if this was self-evident now that AC image versions are decoupled from platform versions? Happy to remove it

Reviewer: @jwitz Yea, I think no need to re-state it. I'd actually just remove this "Support for Latest Builds of AC" section, and bump up a short note on how bug fixes addressed in 2.0.2 were backported to `2.0.0-5` (and then still link "Upgrade Airflow" doc like you do below).

`2.0.2-1` is the "first" release of 2.0.2 and is the same thing as above.

Reviewer 2: This is something transparent to the user as we do not list -patch versions in the UI. And this is something that has existed for more than 3 months now. I vote for not having this section.
@paolaperaza wdyt?

Reviewer: @vishwas-astro I think we should include a note about `2.0.0-5` existing (w/ changelog) since some customers _are_ waiting for bug fixes in there, but I agree it doesn't warrant its own section.

The latest patch versions of existing Astronomer Certified images are now available on all Astronomer platforms on version 0.23.0 or greater:

- [2.0.2-1](https://github.com/astronomer/ap-airflow/blob/master/2.0.2/CHANGELOG.md)
- [2.0.0-5](https://github.com/astronomer/ap-airflow/blob/master/2.0.0/CHANGELOG.md)

> For instructions on how to upgrade to the latest patch version of a release, read [Upgrade Airflow](https://www.astronomer.io/docs/enterprise/v0.23/customize-airflow/manage-airflow-versions).

Reviewer 2: @jwitz now that 2.0.2 is pushed to updates.astronomer.io - we should update the doc https://www.astronomer.io/docs/enterprise/v0.23/customize-airflow/manage-airflow-versions to include 2.0.2 in the list of supported Airflow versions. cc @kaxil

### Bug Fixes

> - Fixed an issue where IAM roles for service accounts ([IRSA](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)) would not connect to S3 back-end registries. ([Source](https://github.com/astronomer/astronomer/pull/1065))

Reviewer 2: The PRs are the same for both bug fixes?

Author: Ope nice catch. Fixed!

Reviewer: Ha whoops also noted this - ignore my comment!

> - Fixed an issue where a Deployment would not work correctly when using the Kubernetes Executor and the KubernetesPodOperator at the same time. ([Source](https://github.com/astronomer/astronomer/pull/1065))

Reviewer: - Looks like the link to this PR is the same one as above (1065)
- I'd try to specify what "a Deployment would not work correctly" means (keep brief obv). What couldn't users do/what wasn't happening?
